src/main/java/core/KitchenSimulator.java

```diff
@@ -0,0 +1,147 @@
+package core;
+
+import controllers.InventoryItemController;
+import controllers.OrderController;
+import controllers.WaiterController;
+import entities.*;
+import managers.OrderQueueManager;
+import models.StatusOrder;
+import utils.IoC;
+import utils.Utils;
+
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class KitchenSimulator {
+
+    private static IoC ioc = IoC.getInstance();
+    private static OrderQueueManager orderQueueManager = new OrderQueueManager();
+
+    public static void main(String[] args) throws InterruptedException {
+        Utils.init();
+
+        KitchenSimulator kitchenSimulator = new KitchenSimulator();
+
+        while (true) {
+            kitchenSimulator.moveOrdersToPendingQueue();
+            kitchenSimulator.executeWaiterJob();
+
+            Thread.sleep(5000);
+        }
+
+    }
+
+    public void moveOrdersToPendingQueue() {
+        var orderController = ioc.<OrderController>getInstance("OrderController");
```

Comment:
Why are you getting `ioc` instances here? why not to declare them a private final available for the whole class when this class is instanciated or initialized?

I think you can do this for all the instances you need.

Also I would suggest a class that will have the responsability to inject all dependencies in the required classes.
Example:
```
class DependenInjector {
   private final Ioc ioc;
   public DependenInjector () {
     this.ioc = IoC.getInstance();
   }
   
   public IoC injectDependencies() {
      this.ioc.put("kitchenSimulator", new KitchenSimulator(this.ioc.get(...)));

       return this.ioc;
   }
}
```